xarray/backends/zarr.py

```diff
@@ -38,9 +40,6 @@
 from xarray.namedarray.utils import module_available
 
 if TYPE_CHECKING:
-    from zarr import Array as ZarrArray
```

Comment:
zarr needs to be an optional import

Reply:
I need zarr types at runtime inside `ZarrStore` -- should I import them locally and raise an import error if someone tries to create a `ZarrStore` without zarr installed?

Reply:
the zarr imports are all now inside the functions that need them